### PR TITLE
Create generate backup dialog

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Create generate backup dialog. Addons support is not implemented yet and will be added later.

CleanShot 2024-11-18 at 17 31 32
CleanShot 2024-11-18 at 17 31 43
CleanShot 2024-11-18 at 17 31 51

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/backup/components/ha-backup-agents-select.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    return html`
+      <div class="agents">
+        ${this.agents.map((agent) => this._renderAgent(agent))}
+      </div>
+    `;
+  }
+
+  private _renderAgent(agent: BackupAgent) {
+    const [domain, name] = agent.agent_id.split(".");
+    const domainName =
```
Comment (bramkragten) [inline src/panels/config/backup/components/ha-backup-agents-select.ts:37]: Use the `domainToName` helper function

Comment (piitaya) [reply src/panels/config/backup/components/ha-backup-agents-select.ts:37]: It's not really a domain like entity domain. I'm also not sure this format will stay.

**Review by bramkragten — APPROVED** [review]

File: src/panels/config/backup/ha-config-backup-dashboard.ts [hunk 2]
```diff
@@ -33,6 +32,10 @@ import { SubscribeMixin } from "../../../mixins/subscribe-mixin";
 import { haStyle } from "../../../resources/styles";
 import type { HomeAssistant, Route } from "../../../types";
 import { brandsUrl } from "../../../util/brands-url";
+import {
+  showAlertDialog,
+  showConfirmationDialog,
+} from "../../lovelace/custom-card-helpers";
```
Comment (bramkragten) [inline src/panels/config/backup/ha-config-backup-dashboard.ts:38]: custom card helpers? 🤔

**Review by bramkragten — CHANGES_REQUESTED** [review]

**Review by bramkragten — APPROVED** [review]

File: src/panels/config/backup/ha-config-backup-dashboard.ts (hunk 2 above)
Comment (piitaya) [reply src/panels/config/backup/ha-config-backup-dashboard.ts]: Automatic import 🤦‍♂️

**Pull request merged**